hypershift-operator/controllers/hostedcluster/internal/platform/core.go

```diff
@@ -0,0 +1,60 @@
+package platform
+
+import (
+	"context"
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/upsert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	hostedClusterAnnotation = "hypershift.openshift.io/cluster"
+)
+
+type Platform interface {
+	// ReconcileCAPIInfraCR is responsible for creating or updating a CAPIInfraCR.
+	ReconcileCAPIInfraCR(hcluster *hyperv1.HostedCluster, name, controlPlaneNamespace string, apiEndpoint hyperv1.APIEndpoint,
+		c client.Client, ctx context.Context) (client.Object, error)
+
+	// CAPIProviderDeploymentSpec should return a CAPIProvider DeploymentSpec.
+	CAPIProviderDeploymentSpec(controlPlaneNamespace, tokenMinterImage string) (*appsv1.DeploymentSpec, error)
+
+	// HCPPlatformSpec should return a hyperv1.PlatformSpec for the HCP.
+	HCPPlatformSpec(hcp *hyperv1.PlatformSpec, controlPlaneNamespace string) error
```

Comment:
This doesn't return anything but an `error`... should it. Seems like it should be returning a `PlatformSpec`, and care should probably be taken to pass a copy (DeepCopy) of the input so implementations can't accidentally mutate the input?

Reply:
Done.

Reply:
Actually I simplified and updated to mirror the full platformSpec from HC to HCP so no assumptions are made about the secret names and any consumer should just look at the API fields. Now this func is dropped.